### PR TITLE
add option to disable the apiserver's insecure port via env var

### PR DESCRIPTION
**Pull request by dekkagaijin**
@mwwolters FYI
/sig gcp

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/manifests/kube-apiserver.manifest [hunk 1]
```diff
@@ -32,17 +32,19 @@
     {{container_env}}
     "livenessProbe": {
       "httpGet": {
+        "scheme": "HTTPS",
         "host": "127.0.0.1",
-        "port": 8080,
+        "port": {{secure_port}},
```
Comment (MrHohn) [inline cluster/gce/manifests/kube-apiserver.manifest]: Does this default to 443 now? Do we allow user to switch back to 8080 and insecure?

Comment (dekkagaijin) [reply cluster/gce/manifests/kube-apiserver.manifest]: It's 443 in our [startup configuration script](https://github.com/kubernetes/kubernetes/blob/ac1557a8861b2c8fe50a2749ea45169a94bd4756/cluster/gce/gci/configure-helper.sh#L1595), but I made it {{secure_port}} here to stay consistent with the other instances in the manifest.

Comment (dekkagaijin) [reply cluster/gce/manifests/kube-apiserver.manifest]: I'm not sure there's a point in using the insecure port for the liveness or readiness probes

Comment (dekkagaijin) [reply cluster/gce/manifests/kube-apiserver.manifest]: I'm guessing that there was also intended to be an {{insecure_port}}, which never got hooked up: https://github.com/kubernetes/kubernetes/blob/ac1557a8861b2c8fe50a2749ea45169a94bd4756/cluster/gce/gci/configure-helper.sh#L1880-L1881

Comment (MrHohn) [reply cluster/gce/manifests/kube-apiserver.manifest]: SG, thanks for digging that up, I was just not clear on the expected behavior.

Comment (MrHohn) [reply cluster/gce/manifests/kube-apiserver.manifest]: BTW should we remove `sed -i -e "s@{{secure_port}}@8080@g" "${src_file}"`? That was a bit confusing :)

**Review by MrHohn — COMMENTED** [review]
/lgtm

File: cluster/gce/manifests/kube-apiserver.manifest (hunk 1 above)
Comment (dekkagaijin) [reply cluster/gce/manifests/kube-apiserver.manifest]: yeah, it's a no-op at best. done.

**Comment by dekkagaijin** [comment]
@mikedanese would you mind approving this one, too?

**Comment by mikedanese** [comment]
Please remember to squash fixup commits.

**Comment by dekkagaijin** [comment]
@mikedanese the second commit was an unrelated 'while I'm in here' fix, didn't think it was appropriate to tie the two commits together

**Pull request merged**